src/flamenco/runtime/fd_executor.c

```diff
@@ -281,32 +294,72 @@ fd_executor_check_txn_data_sz( fd_exec_txn_ctx_t * txn_ctx ) {
 
   ulong accumulated_account_size = 0UL;
 
+  // https://github.com/anza-xyz/agave/blob/eff8961dcf7880aa7e438463203611a8a0088554/svm/src/account_loader.rs#L315-L335
+  for( ulong i=0UL; i<txn_ctx->accounts_cnt; i++ ) {
+    // Check for max loaded acct size
+    fd_borrowed_account_t * acct = NULL;
+    int err        = fd_txn_borrowed_account_view_idx( txn_ctx, (uchar)i, &acct );
+    ulong acc_size = err==FD_ACC_MGR_SUCCESS ? acct->const_meta->dlen : 0UL; 
+  
+    err = accumulate_and_check_loaded_account_data_size( acc_size, requested_loaded_accounts_data_size, &accumulated_account_size );
+    if( FD_UNLIKELY( err!=FD_RUNTIME_EXECUTE_SUCCESS ) ) {
+      return err;
+    }
+  }
+
+  // https://github.com/anza-xyz/agave/blob/ae18213c19ea5335dfc75e6b6116def0f0910aff/svm/src/account_loader.rs#L377-L433
   ushort instr_cnt = txn_ctx->txn_descriptor->instr_cnt;
   for( ushort i=0; i<instr_cnt; i++ ) {
     fd_txn_instr_t  const * instr       = &txn_ctx->txn_descriptor->instr[i];
 
     // https://github.com/anza-xyz/agave/blob/ae18213c19ea5335dfc75e6b6116def0f0910aff/svm/src/account_loader.rs#L384-L386
-    fd_borrowed_account_t * program_acc = NULL;
-    int err = fd_txn_borrowed_account_view_idx( txn_ctx, instr->program_id, &program_acc );
+    fd_borrowed_account_t * program_account = NULL;
+    int err = fd_txn_borrowed_account_view_idx( txn_ctx, instr->program_id, &program_account );
     if( FD_UNLIKELY( err!=FD_ACC_MGR_SUCCESS ) ) {
       return FD_RUNTIME_TXN_ERR_PROGRAM_ACCOUNT_NOT_FOUND;
     }
-  }
 
-  for( ulong i=0UL; i<txn_ctx->accounts_cnt; i++ ) {
+    // https://github.com/anza-xyz/agave/blob/ae18213c19ea5335dfc75e6b6116def0f0910aff/svm/src/account_loader.rs#L387-L389
+    if( FD_UNLIKELY( memcmp( program_account->pubkey->key, fd_solana_native_loader_id.key, sizeof(fd_pubkey_t) ) == 0 ) ) {
+      continue;
+    }
 
-    // Check for max loaded acct size
-    fd_borrowed_account_t * acct = NULL;
-    int err        = fd_txn_borrowed_account_view_idx( txn_ctx, (uchar)i, &acct );
-    ulong acc_size = err==FD_ACC_MGR_SUCCESS ? acct->const_meta->dlen : 0UL; 
-  
-    accumulated_account_size = fd_ulong_sat_add( accumulated_account_size, acc_size );
-  }
+    // https://github.com/anza-xyz/agave/blob/ae18213c19ea5335dfc75e6b6116def0f0910aff/svm/src/account_loader.rs#L397-L400
+    if( FD_UNLIKELY( !program_account->const_meta->info.executable ) ) {
+      return FD_RUNTIME_TXN_ERR_INVALID_PROGRAM_FOR_EXECUTION;
+    }
 
-  if( FD_UNLIKELY( accumulated_account_size>requested_loaded_accounts_data_size ) ) {
-    return FD_RUNTIME_TXN_ERR_MAX_LOADED_ACCOUNTS_DATA_SIZE_EXCEEDED;
+    // https://github.com/anza-xyz/agave/blob/ae18213c19ea5335dfc75e6b6116def0f0910aff/svm/src/account_loader.rs#L402-L405
+    // No FD_UNLIKELY or FD_LIKELY since it holds true for all native program invocations
+    if( memcmp( program_account->const_meta->info.owner, fd_solana_native_loader_id.key, sizeof(fd_pubkey_t) ) == 0 ) {
```

Comment:
lets do `!memcmp( program_account->const_meta->info.owner, fd_solana_native_loader_id.key, sizeof(fd_pubkey_t) )` instead of remove spaces around the `==`